src/shared/watchdog.c

```diff
@@ -114,9 +117,37 @@ int watchdog_set_timeout(usec_t *usec) {
         return r;
 }
 
+usec_t watchdog_runtime_wait(void) {
+        usec_t rtwait = USEC_INFINITY;
+        usec_t ntime;
+
+        if (timestamp_is_set(watchdog_timeout)) {
+                if (timestamp_is_set(watchdog_last_ping)) {
+                        ntime = now(clock_boottime_or_monotonic());
+                        assert(ntime >= watchdog_last_ping);
+                        rtwait = watchdog_last_ping + (watchdog_timeout / 2) - ntime;
```

Comment:
this will underflow if for some reason (some scheduling latency) we didn't get scheduled quick enough as in that case we the half timeout already elapsed...

usec_t is unsigned after all.. please catch this explicitly before doing the subtraction here... or just use `usec_sub_unsigned()` which does this for you